src/traversals/randomwalks.jl

```diff
@@ -17,6 +17,68 @@ function randomwalk(g::SimpleGraph, s::Integer, niter::Integer)
   return visited[1:i-1]
 end
 
+"""Performs a non-backtracking random walk on graph `g` starting at vertex `s` and continuing for
+a maximum of `niter` steps. Returns a vector of vertices visited in order.
+"""
+function non_backtracking_randomwalk(g::Graph, s::Integer, niter::Integer)
+    s in vertices(g) || throw(BoundsError())
+    visited = Vector{Int}()
+    sizehint!(visited, niter)
+    currs = s
+    prev = -1
+    i = 1
+
+    push!(visited, currs)
+    i += 1
+    nbrs = out_neighbors(g,currs)
+    length(nbrs) == 0 && return visited[1:i-1]
+    prev = currs
+    currs = rand(nbrs)
+
+    while i <= niter
+        push!(visited, currs)
+        i += 1
+        nbrs = out_neighbors(g,currs)
+        length(nbrs) == 1 && break
+        idnext = rand(1:length(nbrs)-1)
```

Comment:
I like the rand(nbrs) in DiGraph. Why are we doing this differently here, and can we change the other functions as well?